test/oqs_test_evp_pkey_params.c

```diff
@@ -0,0 +1,613 @@
+ *
+ * \param ctx EVP_PKEY context.
+ *
+ * \returns 0 on success. */
+static int init_keygen(EVP_PKEY_CTX *ctx)
+{
+    int err;
+
+    if ((err = EVP_PKEY_keygen_init(ctx)) == -2) {
+        fputs("`EVP_PKEY_keygen_init` failed, couldn't initialize keygen: not "
+              "supported\n",
+              stderr);
+    } else if (err <= 0) {
+        fputs("`EVP_PKEY_keygen_init` failed, couldn't initialize keygen: ",
+              stderr);
+        ERR_print_errors_fp(stderr);
+        fputc('\n', stderr);
+    }
+
+    return err;
+}
+
+/** \brief Generates the private key.
+ *
+ * \param ctx EVP_PKEY context.
+ *
+ * \returns The private key, or `NULL` if an error occurred. */
+static EVP_PKEY *generate_private_key(EVP_PKEY_CTX *ctx)
+{
+    EVP_PKEY *private_key = NULL;
+    int err;
+
+    if ((err = EVP_PKEY_generate(ctx, &private_key)) == -2) {
+        fputs("`EVP_PKEY_generate` failed, couldn't generate: not supported\n",
+              stderr);
+    } else if (err <= 0) {
+        fputs("`EVP_PKEY_generate` failed, couldn't generate: ", stderr);
+        ERR_print_errors_fp(stderr);
+        fputc('\n', stderr);
+    }
+
+    return private_key;
+}
+
+/** \brief Extracts an octet string from a parameter of an EVP_PKEY.
+ *
+ * \param key The EVP_PKEY;
+ * \param param_name Name of the parameter.
+ * \param[out] buf Out buffer.
+ * \param[out] buf_len Size of out buffer.
+ *
+ * \returns 0 on success. */
+static int get_param_octet_string(const EVP_PKEY *key, const char *param_name,
+                                  uint8_t **buf, size_t *buf_len)
+{
+    *buf = NULL;
+    *buf_len = 0;
+    int ret = -1;
+
+    if (EVP_PKEY_get_octet_string_param(key, param_name, NULL, 0, buf_len)
+        != 1) {
+        fprintf(stderr,
+                "`EVP_PKEY_get_octet_string_param` failed with param `%s`: ",
+                param_name);
+        ERR_print_errors_fp(stderr);
+        fputc('\n', stderr);
+        goto out;
+    }
+    if (!(*buf = malloc(*buf_len))) {
+        fprintf(stderr, "failed to allocate %#zx byte(s)\n", *buf_len);
+        goto out;
+    }
+    if (EVP_PKEY_get_octet_string_param(key, param_name, *buf, *buf_len,
+                                        buf_len)
+        != 1) {
+        fprintf(stderr,
+                "`EVP_PKEY_get_octet_string_param` failed with param `%s`: ",
+                param_name);
+        ERR_print_errors_fp(stderr);
+        fputc('\n', stderr);
+        free(*buf);
+        *buf = NULL;
+    } else {
+        ret = 0;
+    }
+
+out:
+    return ret;
+}
+
+/** \brief Extracts the classical keys from an hybrid key.
+ *
+ * \param private_key The private key.
+ * \param[out] out Key pair where to write the keys.
+ *
+ * \returns 0 on success. */
+static int private_key_params_get_classical_keys(const EVP_PKEY *private_key,
+                                                 struct KeyPair *out)
+{
+    int ret = -1;
+
+    if (get_param_octet_string(private_key, "hybrid_classical_pub",
```

Comment:
Why not reference the 4 new PARAM constants as such? Shouldn't they be in some header file, probably `oqs_prov.h`?